### ISSUE TITLE
Introduce 'null notation' for placeholder selectors

### ISSUE THREAD

**Issue by axunonb** [axuno/SmartFormat#157]
follow-up on #156
```csharp
class Address
{
    public string Country;
    public string City;
}

class Person
{
    public string FirstName = "first";
    public string LastName = "last";
    public Address? Address;
}
```
`nullable` notation like `Smart.Format("{Person?.Address.City}", new Person());`

@mikasoukhov:

1. With "ignore" you mean, `Person?.Address.City` should **render as `string.Empty`** if `Address` is `null`? Anything else in the format string should render.
2. And `Person.?Address` should **not throw** if `Address` is `null` and the selector is marked as `nullable`, but it **should** throw if not `nullable` and `ErrorAction.ThrowError` is set?

**Comment by axunonb** [comment]
Null notation will be introduced in version 3, and it is implemented with #176 in the way described above.
Any feedback welcome.

**Comment by axunonb** [comment]
@mikasoukhov:
For your convenience here is a nuget package with the current v3 binaries (renamed as .zip).
Would be great if you could have a closer look at the `nullable` implementation. There is also a [short description](https://github.com/axuno/SmartFormat/pull/176).
[SmartFormat.NET.3.0.0-pre-alpha.nupkg.zip](https://github.com/axuno/SmartFormat/files/6673371/SmartFormat.NET.3.0.0-pre-alpha.nupkg.zip)

**Comment by mikasoukhov** [comment]
Hello

1. null or empty string (this it equals for most cases).
2. think so.

**Comment by axunonb** [comment]
Thanks, that's exactly how it is implemented in v3 (#176) now.
